detox/src/utils/exec.js

```diff
@@ -139,9 +140,18 @@ function spawnAndLog(command, flags, options) {
     log.error({ event: 'SPAWN_ERROR' }, `${cmd} failed with code = ${exitCode}`);
   }
 
-  if (!options || !options.silent) {
-    stdout && stdout.on('data', (chunk) => log.trace({ stdout: true, event: 'SPAWN_STDOUT' }, chunk.toString()));
-    stderr && stderr.on('data', (chunk) => log.error({ stderr: true, event: 'SPAWN_STDERR' }, chunk.toString()));
+  if (!silent) {
+    stdout && stdout.on('data', (chunk) => {
+      const line = chunk.toString();
+      const loglevel = inferLogLevel(line, logLevelPatterns) || 'trace';
+      log[loglevel]({ stdout: true, event: 'SPAWN_STDOUT' }, line);
+    });
+
+    stderr && stderr.on('data', (chunk) => {
+      const line = chunk.toString();
+      const loglevel = inferLogLevel(line, logLevelPatterns) || 'error';
+      log[loglevel]({ stderr: true, event: 'SPAWN_STDERR' }, line);
+    });
```

Comment:
@noomorph since this is "real-time"-ish, may I suggest to preregister different handler-callbacks that do not check for the value of `logLevelPatterns` on each `chunk`, in case it's empty.

Reply:
Checked a good couple of times with profiler, etc — no impact here on the performance. Not sure if it really makes sense 🤷‍♂️

Reply:
@d4vidi , it is exiting very fast there, see the source code: https://github.com/lodash/lodash/blob/master/isEmpty.js?source=post_page---------------------------#L47 there. Function calls themselves are quite cheap. `chunk.toString()` takes way more time.

Reply:
Yes I understand - but still, no point in checking again and again whether it's empty because it never changes through-out the life of the callback...